anvil/src/main/kotlin/trikita/anvil/CustomDslSetter.kt

```diff
@@ -14,9 +14,12 @@ import trikita.anvil.*
 import java.util.*
 
 // weight constants
-const val FILL = ViewGroup.LayoutParams.MATCH_PARENT
-const val MATCH = ViewGroup.LayoutParams.MATCH_PARENT
-const val WRAP = ViewGroup.LayoutParams.WRAP_CONTENT
+sealed class Size(val value: Int) {
+    object FILL : Size(ViewGroup.LayoutParams.MATCH_PARENT)
+    object MATCH : Size(ViewGroup.LayoutParams.MATCH_PARENT)
+    object WRAP : Size(ViewGroup.LayoutParams.WRAP_CONTENT)
+    class DIP(value: Int) : Size(value)
+}
```

Comment:
I guess we can remove `FILL` constant entirely.

Is there a chance we can avoid splitting sizes into `sealed class Size` and `Px` / `Sp` / `Dip` mentioned below? `MATCH` and `WRAP` would just be `Px` instances. We don't gain much from having them `inline` anyway as type checking in setters would still box them if I understand the rules correctly.

Reply:
Hmm, I am not really sure about making MATCH and WRAP a Px instance, if I understood you correctly. IMO the signature of methods won't express the semantics. For example if size(w : Px, h : Px) then it is not obvious that you can pass MATCH or WRAP constants there.

Reply:
Yeah, I guess you're right. It's also not reasonable to allow `MATCH` and `WRAP` everywhere size is expected, so they are a different beast needed mostly in layout params.
I'm still a bit concerned about separated `Size.DIP` and `Dip`.

Reply:
Yeah, I agree. Ideally they would be the same, and it will allow to drop dpSize() extension, but so far I don't see how a way how it can be implemented in different way. I am open to suggestions)

Reply:
Awesome, let's land on your latest changes and move on 😊